packages/contracts/contracts/optimistic-ethereum/libraries/trie/Lib_MerkleTrie.sol

```diff
@@ -348,7 +348,32 @@ library Lib_MerkleTrie {
         TrieNode[] memory newNodes = new TrieNode[](3);
         uint256 totalNewNodes = 0;
 
-        if (keyRemainder.length == 0 && lastNodeType == NodeType.LeafNode) {
+        // Reference: https://github.com/ethereumjs/merkle-patricia-tree/blob/c0a10395aab37d42c175a47114ebfcbd7efcf059/src/baseTrie.ts#L294-L313
+        bool matchLeaf = false;
+        if (lastNodeType == NodeType.LeafNode) {
+            uint256 l = 0;
+            if (_path.length > 0) {
+                for (uint256 i = 0; i < _path.length - 1; i++) {
+                    if (_getNodeType(_path[i]) == NodeType.BranchNode) {
+                        l++;
+                    } else {
+                        l += _getNodeKey(_path[i]).length;
+                    }
+                }
+            }
+
+            if (
+                _getSharedNibbleLength(
+                    _getNodeKey(lastNode),
+                    Lib_BytesUtils.slice(Lib_BytesUtils.toNibbles(_key), l)
+                ) == _getNodeKey(lastNode).length
+                && keyRemainder.length == 0
+            ) {
+                matchLeaf = true;
+            }
+        }
+
+        if (matchLeaf) {
```

Comment:
Was there a bug here?

Reply:
Yeah, we had a bug here that could potentially have caused a fraud proof to break